### DISCUSSION TITLE
Need help to build: IOS 14 support + universal IPA(32bit and 64bit)

### DISCUSSION THREAD

**Discussion by Dima107** [airsdk/Adobe-Runtime-Support#615] [Q&A]
MacOS 10.14.6 (Mojave) (Higher versions don't support 32 bit)
XCode 10.3.1 (the highest version for Mojave)
AIR SDK 33.1.1.300 (and 98)
Also I tried to use custom IOS SDKs: 14.0 and 14.1
Intellij IDEA 2020.3

we use Ironsource mediation 
https://developers.ironsrc.com/ironsource-mobile/air/adobe-air-plugin/

I can't build project with Ironsource Facebook Adapter and FAN SDK. I've got build error:
_ld: in /var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork(FBANInterstitialAdAdapter.m.o), could not parse object file /var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork(FBANInterstitialAdAdapter.m.o): 'Invalid bitcode version (Producer: '1103.0.32.62.0_0' Reader: '1100.0.33.17_0')', using libLTO version 'LLVM version 11.0.0, (clang-1100.0.33.17)' file '/var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork' for architecture arm64
ld: in /var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork(FBANInterstitialAdAdapter.m.o), could not parse object file /var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork(FBANInterstitialAdAdapter.m.o): 'Invalid bitcode version (Producer: '1103.0.32.62.0_0' Reader: '1100.0.33.17_0')', using libLTO version 'LLVM version 11.0.0, (clang-1100.0.33.17)' file '/var/folders/x3/f5fsj0y55zz44_pnzndm0bwc0000gn/T/17110a0f-c63c-42e5-abe9-6b5d6d019fb5/FBAudienceNetwork.framework/FBAudienceNetwork' for architecture armv7
Compilation failed while executing : ld64_

I built same build from macOS Big Sur successfully. But it was 64 bit build only. But we need universal build

Also I tried to build from Windows (windows 10, air sdk 33.1.1.300). I've got error
_ld: file is universal (4 slices) but does not contain the arm64 architecture: C:\\Users\\Dmytro\\AppData\\Local\\Temp\\9980dfce-12c4-467d-854f-98b60cf1e90b/FBAudienceNetwork.framework/FBAudienceNetwork for architecture arm64
ld: file is universal (4 slices) but does not contain the armv7 architecture: C:\\Users\\Dmytro\\AppData\\Local\\Temp\\9980dfce-12c4-467d-854f-98b60cf1e90b/FBAudienceNetwork.framework/FBAudienceNetwork for architecture armv7
Compilation failed while executing : ld64_
If I use custom IOS SDK:
_ld: unexpected token: !tapi-tbd for architecture arm64
ld: unexpected token: !tapi-tbd for architecture armv7
Compilation failed while executing : ld64_

Can somebody help me with it?

**Comment by amorganiv** [comment]
I'm seeing this same error output using AIR SDK 33.1.300 & iPhoneOS14.1.sdk:
ld: unexpected token: !tapi-tbd for architecture armv7
ld: unexpected token: !tapi-tbd for architecture arm64
Compilation failed while executing : ld64

Also, I'm using the distriqt/ANE-InAppBilling. The app builds correctly using AIR SDK 33.1.1.98 & iPhoneOS13.1.sdk
